cpp/CMakeLists.txt

```diff
@@ -64,41 +64,56 @@ if(KvikIO_REMOTE_SUPPORT)
   include(cmake/thirdparty/get_libcurl.cmake)
 endif()
 
-if(KvikIO_CUDA_SUPPORT)
-  rapids_find_package(
-    CUDAToolkit REQUIRED
-    BUILD_EXPORT_SET kvikio-exports
-    INSTALL_EXPORT_SET kvikio-exports
-  )
-  include(cmake/thirdparty/get_nvtx.cmake)
-endif()
-
 rapids_find_package(
-  cuFile
+  CUDAToolkit
   BUILD_EXPORT_SET kvikio-exports
   INSTALL_EXPORT_SET kvikio-exports
 )
 
-if(NOT cuFile_FOUND)
+if(KvikIO_CUDA_SUPPORT)
+  if(NOT CUDAToolkit_FOUND)
+    message(FATAL_ERROR "CUDA support requested but CUDA toolkit not found")
+  endif()
+
+  include(cmake/thirdparty/get_nvtx.cmake)
+endif()
+
+if(NOT TARGET CUDA::cuFile)
```

Comment:
I think this whole block can go inside the `if(KvikIO_CUDA_SUPPORT)` block, right? You never check any of this if you aren't building for CUDA support.

Reply:
Is the CUDA support different from the cuFile support?

Reply:
cuFile support requires CUDA support to be enabled.